packages/neuron-wallet/src/block-sync-renderer/sync/light-connector.ts

```diff
@@ -98,52 +85,92 @@ export default class LightConnector extends Connector<CKBComponents.Hash> {
   }
 
   private async synchronize() {
-    if (!this.syncQueue.idle()) {
-      return
+    const syncScripts = await this.upsertTxHashes()
+    await this.updateSyncScriptBlock(syncScripts)
+    const minSyncBlockNumber = await SyncProgressService.getCurrentWalletMinBlockNumber()
+    const hasNextBlock = await this.notifyAndSyncNext(minSyncBlockNumber)
+    if (!hasNextBlock) {
+      await this.updateBlockStartNumber(minSyncBlockNumber)
+    }
+  }
+
+  private async getTransactions({
+    script,
+    blockRange,
+    scriptType,
+  }: {
+    script: Script
+    blockRange: [HexString, HexString]
+    scriptType: ScriptType
+  }) {
+    let result = await this.lightRpc.getTransactions(
+      { script, blockRange, scriptType },
+      'asc',
+      '0x64',
+      undefined as unknown as HexString
+    )
+    if (!result.txs.length) {
+      return []
     }
-    await this.subscribeSync()
+    const res = [...result.txs]
+    while (result.lastCursor !== '0x') {
```

Comment:
The first request could be moved into the while loop with condition `while(!['0x', undefined].includes(lastCursor))`

Reply:
May `lastCursor === undefined || lastCursor !== '0x'`

Reply:
Sure, `lastCursor != '0x'` is enough here